File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> out(nums.size(), 1);

        int pre = 1;
        for (int i = 0; i < nums.size(); ++i) {
            out[i]*=pre;
            pre*=nums[i];
        }

        int post = 1;
        for (int i = nums.size() - 1; i >= 0; --i) {
            out[i]*=post;
            post*=nums[i];
        }

        return out;

    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        tes = set(nums)
        longest = 0
        for num in nums:
            if num - 1 not in tes:
                off = 0
                while (True):
                    if num + off not in tes: break
                    off += 1

                longest = off if off > longest else longest
        return longest